src/main/java/gregtech/common/covers/filter/SimpleItemFilter.java

```diff
@@ -82,38 +65,95 @@ public boolean showGlobalTransferLimitSlider() {
     }
 
     @Override
-    public int getTotalOccupiedHeight() {
-        return 36;
-    }
-
-    @Override
-    public void initUI(Consumer<Widget> widgetGroup) {
+    public void initUI(Consumer<gregtech.api.gui.Widget> widgetGroup) {
         for (int i = 0; i < 9; i++) {
-            widgetGroup.accept(new PhantomSlotWidget(itemFilterSlots, i, 10 + 18 * (i % 3), 18 * (i / 3))
+            widgetGroup.accept(new PhantomSlotWidget(filterReader, i, 10 + 18 * (i % 3), 18 * (i / 3))
                     .setBackgroundTexture(GuiTextures.SLOT));
         }
         widgetGroup.accept(new ToggleButtonWidget(74, 0, 20, 20, GuiTextures.BUTTON_FILTER_DAMAGE,
-                () -> ignoreDamage, this::setIgnoreDamage).setTooltipText("cover.item_filter.ignore_damage"));
+                filterReader::isIgnoreDamage, filterReader::setIgnoreDamage)
+                        .setTooltipText("cover.item_filter.ignore_damage"));
         widgetGroup.accept(new ToggleButtonWidget(99, 0, 20, 20, GuiTextures.BUTTON_FILTER_NBT,
-                () -> ignoreNBT, this::setIgnoreNBT).setTooltipText("cover.item_filter.ignore_nbt"));
+                filterReader::isIgnoreNBT, filterReader::setIgnoreNBT).setTooltipText("cover.item_filter.ignore_nbt"));
     }
 
     @Override
-    public void writeToNBT(NBTTagCompound tagCompound) {
-        tagCompound.setTag("ItemFilter", itemFilterSlots.serializeNBT());
-        tagCompound.setBoolean("IgnoreDamage", ignoreDamage);
-        tagCompound.setBoolean("IgnoreNBT", ignoreNBT);
+    public @NotNull ModularPanel createPopupPanel(GuiSyncManager syncManager) {
+        return GTGuis.createPopupPanel("simple_item_filter", 98, 81)
+                .child(CoverWithUI.createTitleRow(getContainerStack()))
+                .child(createWidgets(syncManager).top(22).left(4));
+    }
+
+    @Override
+    public @NotNull ModularPanel createPanel(GuiSyncManager syncManager) {
+        return GTGuis.createPanel("simple_item_filter", 176, 166);
+    }
+
+    @Override
+    public @NotNull Widget<?> createWidgets(GuiSyncManager syncManager) {
+        SlotGroup filterInventory = new SlotGroup("filter_inv", 3, 1000, true);
+        var ignoreDamage = new BooleanSyncValue(this.filterReader::isIgnoreDamage, this.filterReader::setIgnoreDamage);
+        var ignoreNBT = new BooleanSyncValue(this.filterReader::isIgnoreNBT, this.filterReader::setIgnoreNBT);
```

Comment:
Why `var` when you know what these will be?

Reply:
idk, typing three letters was faster. i could change it back to be explicitly typed.

Reply:
Its OK. I was just wondering